avm/src/vm/opcode.rs

```diff
@@ -2758,48 +2758,77 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
     })
   });
 
-  async fn __httpget(uri: String) -> std::result::Result<String, Box<dyn std::error::Error>> {
-    let url = TryInto::<Uri>::try_into(uri)?;
-    let mut response = Client::builder()
+  fn __httpreq(
+    method: String,
+    uri: String,
+    headers: Vec<(String, String)>,
+    body: Option<String>,
+  ) -> ResponseFuture {
+    let mut req = Request::builder()
+      .method(method.as_str())
+      .uri(uri.as_str());
+    for header in headers {
+      req = req.header(header.0.as_str(), header.1.as_str());
+    }
+    let req_obj = if body.is_some() {
+      req.body(Body::from(body.unwrap())).unwrap()
+    } else {
+      req.body(Body::empty()).unwrap()
+    };
```

Comment:
assuming our code is perfect, if `req.body(...)` returns an `Err` we should probably give that error to the user, since this is an issue with input and can probably be resolved.